code/modules/mob/mob_helpers.dm

```diff
@@ -73,6 +73,10 @@ mob/proc/remove_internal_organ()
 		return .
 	else if(has_reagent_in_blood(DETCOFFEE))
 		return NOIRMATRIX
+	else if(get_item_by_slot(slot_glasses) && istype(get_item_by_slot(slot_glasses), /obj/item/clothing/glasses/scanner))
```

Comment:
don't call get_item_by_slot 3 times.
use `is_wearing_item` with the appropriate parameters and store the result in a variable, run your checks on that